include/dr/mhp/containers/distributed_dense_matrix.hpp

```diff
@@ -0,0 +1,323 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "index.hpp"
+#include "segments.hpp"
+#include "rows.hpp"
+#include "subrange.hpp"
+
+namespace dr::mhp {
+
+using key_type = index<>;
+
+class distributed_matrix_partition {};
+class by_row final : public distributed_matrix_partition {};
+class by_column final : public distributed_matrix_partition {};
+class tiled final : public distributed_matrix_partition {};
+
+template <typename DM> class dm_segments : public std::span<d_segment<DM>> {
```

Comment:
remove this class, it is not needed as you may use rng::all view like distributed_vector does, and it is also dangerous as it not derives from `rng::view_base`